File: src/GameLoop.hs
{-# LANGUAGE OverloadedStrings #-}
module GameLoop where

import Data.Time

main :: IO ()
main = do
  --init functions here
  gameLoop

gameLoop :: IO ()
gameLoop = do
  start <- getCurrentTime
  --input
  --update
  --render
  delay' start 
  gameLoop

delay' :: UTCTime -> IO ()
delay' start = do 
  end <- getCurrentTime
  if diffUTCTime end start < 1.0/60
  then delay' start 
  else return ()

File: src/GameLoop.java

public class GameLoop {

  public static void main(String[] args) {
    init();
    gameLoop();
  }

  static void init() {}
  static void gameLoop() {
    long frameStart, frameTime, delay;
    final long FPS = 1000 / 60;
    while (true) {
      frameStart = System.currentTimeMillis();
      // inputs();
      // update();
      // render();
      delay = FPS - (System.currentTimeMillis() - frameStart);
      try {
        Thread.sleep(delay);
      } catch (Exception e) {
        e.printStackTrace();
      }
    }
  }
}

